autospec/tarball.py

```diff
@@ -332,15 +333,22 @@ def name_and_version(name_arg, version_arg, filemanager):
         github_patterns = [r"https?://github.com/(.*)/(.*?)/archive/[v|r]?.*/(.*).tar",
                            r"https?://github.com/(.*)/(.*?)/archive/[-a-zA-Z]*-(.*).tar",
                            r"https?://github.com/(.*)/(.*?)/archive/[vVrR]?(.*).tar",
-                           r"https?://github.com/(.*)/(.*?)/releases/download/v.*/(.*).tar"]
+                           r"https?://github.com/(.*)/(.*?)/releases/download/.*?/(.*).tar",
+                           r"https?://github.com/(.*)/(.*?)/files/.*?/(.*).tar"]
 
         for pattern in github_patterns:
             m = re.search(pattern, url)
             if m:
-                name = m.group(2).strip()
+                repo = m.group(2).strip()
+                if repo not in name:
```

Comment:
How do we know this is more descriptive if it is different?

Reply:
It's only an assumption, but it holds for the current `test/packageurls`.

Reply:
More specifically, it's a replacement for the prior behavior of _always_ picking the repo name over the tarball-derived name.